### PR TITLE
chore: add logs for missing messages - WPB-9221

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9221" title="WPB-9221" target="_blank">BugWPB-9221</a>  Missing messages during sync
  </td></table>
  <br />
 

### Issue

* remove extra log of CoreCrypto error -> propagate error to the caller with associated value
* fix storing last event id when decryption error occurs
* move eventProcessing logs to WireLogger

### Testing

Updated current tests

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift [hunk 1]
```diff
@@ -301,7 +305,11 @@ extension EventDecoder {
             )
         }
 
-        self.eventMOC.saveOrRollback()
+        do {
+            try self.eventMOC.save()
+        } catch {
+            WireLogger.updateEvent.critical("Failed to save stored update events: \(error.localizedDescription)")
+        }
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift:312]: changed to log error if we cannot save event db

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift [hunk 2]
```diff
@@ -193,6 +193,10 @@ extension EventDecoder {
     ) async -> [ZMUpdateEvent] {
         let decryptedEvents = await decryptEvent(event: event, publicKeys: publicKeys, proteusService: proteusService)
 
+        guard decryptedEvents.isNonEmpty else {
+            return []
+        }
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift:198]: fix storing the event id in case of decryption error

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift [hunk 3]
```diff
@@ -35,6 +35,7 @@ extension EventDecoder {
         WireLogger.updateEvent.info("decrypting proteus event...", attributes: event.logAttributes)
 
         guard !event.wasDecrypted else {
+            WireLogger.updateEvent.info("returned already decrypted event", attributes: event.logAttributes)
             return event
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift:39]: I don't know in which case this would happen

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift (hunk 2 above)
Comment (typfel) [reply wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift:198]: Why shouldn't we store the event id if there's a decryption error?

Comment (netbe) [reply wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift:198]: to avoid duplicate decryption error

**Pull request merged**